src/time/Waveform.hpp

```diff
@@ -16,12 +16,19 @@ class Waveform {
   friend class testing::WaveformFixture; // Make the fixture friend of this class
 public:
   /**
-   * @brief Waveform object which stores values of current and past time windows for performing extrapolation.
-   * @param valuesSize defines how many values one sample in time consists of
+   * @brief Waveform object which stores values of current and past time windows for performing extrapolation. 
+   *
+   * Storage still needs to be initialized with Waveform::initializeData, before the Waveform can be used.
+   *
    * @param extrapolatioOrder defines the maximum extrapolation order supported by this Waveform and reserves storage correspondingly
    */
-  Waveform(const int valuesSize,
-           const int extrapolationOrder);
+  Waveform(const int extrapolationOrder);
+
+  /**
+   * @brief Used to initialize _timeWindowsStorage according to required size.
+   * @param valuesSize defines how many values one sample in time consists of
+   */
+  void initializeData(const int valuesSize);
```

Comment:
The name indicates that data is actually initialized here. In preCICE words, this means that data is set to non-zero values. Why not just `initialize`? Or `initializeStorage`?

Reply:
`initialize` is good. In case we have to initialize more things at a later point in time, we also don't have to change it again.